net/dnscache/dnscache.go

```diff
@@ -278,10 +278,10 @@ func (r *Resolver) lookupIP(host string) (ip, ip6 netip.Addr, allIPs []netip.Add
 
```

Comment:
so at this point I think we should clean `ips` of all `Is4In6` entries. Currently these "polluted" entries leak into the `r.addIPCache` on line 295 where they stick around for other callers, complicating the.

I'm involved in Go discussions about all of this currently. Hopefully we'll later fix up LookupNetIP, etc.

For now let's just do some cleanup pass here, like:

```go
      // Unmap everything; LookupNetIP can return mapped addresses (see #5698)
      for i := range ips {
           ips[i] = ips[i].Unmap()
      }
```